src/MarginTrading.SettingsService/Controllers/AssetPairsController.cs

```diff
@@ -103,15 +104,35 @@ public async Task<AssetPairContract> Insert([FromBody] AssetPairContract assetPa
             }
 
             await _eventSender.SendSettingsChangedEvent($"{Request.Path}", SettingsChangedSourceType.AssetPair);
+            await _cqrsMessageSender.SendAssetPairChangedEvent(new AssetPairChangedEvent
+            {
+                OperationId = Guid.NewGuid().ToString("N"),
+                AssetPair = assetPair,
+            });
             
             return assetPair;
         }
 
+        /// <summary>
+        /// Create new asset pairs in a batch request
+        /// </summary>
+        [HttpPost]
+        [Route("batch")]
+        public async Task<List<AssetPairContract>> BatchInsert([FromBody] AssetPairContract[] assetPairs)
+        {
+            var result = new List<AssetPairContract>();
+            
+            foreach (var assetPair in assetPairs)
+            {
+                result.Add(await Insert(assetPair));
```

Comment:
Partial execution is possible here and it is not good..
I think we should use batch is methods in azure storage repo and in sql repo. 
In both cases operation will be performed in single transaction.